packages/api/test/user.spec.js

```diff
@@ -209,6 +209,80 @@ describe('GET /user/uploads', () => {
     assert.deepStrictEqual(uploads, [...userUploads].sort((a, b) => b.name.localeCompare(a.name)))
   })
 
+  it('lists uploads sorted by date', async () => {
+    const token = await getTestJWT()
+    const res = await fetch(new URL('/user/uploads?sortBy=Date', endpoint).toString(), {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${token}` }
+    })
+    assert(res.ok)
+    const uploads = await res.json()
+    assert.deepStrictEqual(uploads, [...userUploads].sort((a, b) => b.created.localeCompare(a.created)))
+  })
+
+  it('lists uploads in reverse order when sorting by Asc', async () => {
+    const token = await getTestJWT()
+    const res = await fetch(new URL('/user/uploads?sortBy=Name&sortOrder=Asc', endpoint).toString(), {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${token}` }
+    })
+
+    assert(res.ok)
+
+    const uploads = await res.json()
+    const sortedUploads = [...userUploads].sort((a, b) => a.name.localeCompare(b.name))
+
+    assert.deepStrictEqual(uploads, sortedUploads)
+  })
+
+  it('filters results by before date', async () => {
+    const token = await getTestJWT()
+
+    const beforeFilterDate = new Date('2021-07-10T00:00:00.000000+00:00').toISOString()
+    const res = await fetch(new URL(`/user/uploads?before=${beforeFilterDate}`, endpoint).toString(), {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${token}` }
+    })
+
+    assert(res.ok)
+
+    const uploads = await res.json()
+
+    assert(uploads.length < userUploads.length, 'Ensure some results are filtered out.')
+    assert(uploads.length > 0, 'Ensure some results are returned.')
+
+    // Filter uploads fixture by the filter date.
+    const uploadsBeforeFilterDate = userUploads.filter((upload) => {
+      return upload.created <= beforeFilterDate
+    })
```

Comment:
I generally don't like this kind of logic in tests, and perhaps could simply expect something like 3 results...
But seems like these fixtures are shared across tests, so didn't want this test to break when another result is added in the uploads fixture. 
Thoughts on this?

Reply:
I think that the way you've done this is probably the best way given that we're using fixtures that are defined elsewhere. Like you say, we don't want the test to break (or even worse, the code to break and the test to not catch it) if/when someone changes the fixtures.

I think that the real solution to this is for us to move away from using fixtures for tests, and to instead explicitly create the objects that we want within each test. We're starting to do this in places. This is probably a good example (creating a user tag via the DB client and then testing the API endpoint that fetches that info): https://github.com/web3-storage/web3.storage/blob/main/packages/api/test/user.spec.js#L52-L65

Reply:
Thanks! I totally agree, moving away from fixtures makes a huge amount of sense.
I think I'll leave this for now, so the upload tests can all use fixtures consistently, but it would make sense to update this at some point!